test/test_process_template.rb

```diff
@@ -0,0 +1,44 @@
+require 'test_helper'
+
+# Process Template tests
+class TestProcessTemplate < MiniTest::Test
+  def test_process_template
+    client = Kubeclient::Client.new 'http://localhost:8080/api/', 'v1'
+    template = {}
+    template[:metadata] = {}
+    template[:metadata][:name] = 'my-template'
+    template[:metadata][:namespace] = 'default'
+    template[:kind] = 'Template'
+    template[:apiVersion] = 'v1'
+    service = {}
+    service[:metadata] = {}
+    service[:metadata][:name] = '${NAME_PREFIX}my-service'
+    service[:kind] = 'Service'
+    service[:apiVersion] = 'v1'
+    template[:objects] = [service]
+    param = { name: 'NAME_PREFIX', value: 'test/' }
+    template[:parameters] = [param]
+
+    req_body = "{\"metadata\":{\"name\":\"my-template\",\"namespace\":\"default\"},"\
+    "\"kind\":\"Template\",\"apiVersion\":\"v1\",\"objects\":[{\"metadata\":"\
+    "{\"name\":\"${NAME_PREFIX}my-service\"},\"kind\":\"Service\",\"apiVersion\":\"v1\"}],"\
+    "\"parameters\":[{\"name\":\"NAME_PREFIX\",\"value\":\"test/\"}]}"
```

Comment:
Is this the best way to create the request body?
Do we really need a request body here? I see that in other tests we just mocked without this and it seems to be enough.